core/reporter-impl/src/test/java/org/arquillian/reporter/impl/builder/AllBuildersTest.java

```diff
@@ -0,0 +1,134 @@
+
+        AbstractReport report = reportClass.newInstance();
+        report.setName(new UnknownStringKey("report"));
+
+        Map<String, String> mapOfStrings = ImmutableMap.of("Map_Key_1", "Map_Value_1", "Map_Key_2", "Map_Value_2");
+
+        ReportBuilder reportBuilder = Reporter.createReport(report)
+                .addEntry("entry")
+                .addEntry(new StringEntry("String Entry"))
+                .addEntry(new KeyValueEntry("Key Entry", "Value Entry"))
+                .addEntry(new FileEntry("file path"))
+                .addEntries(Arrays.asList(new StringEntry("list_entries_1"), new StringEntry("list_entries_2")))
+                .addEntries(new String[]{"array_strings_1", "array_strings_1"})
+                .addEntries(new StringEntry[]{new StringEntry("array_entries_1"), new StringEntry("array_entries_2")})
+                .addKeyValueEntry("key", "value")
+                .feedKeyValueListFromMap(mapOfStrings)
+                .addKeyValueEntry(DummyStringKeys.DUMMY_1_NAME, "value")
+                .addKeyValueEntry(DummyStringKeys.DUMMY_2_NAME, 1)
+                .addKeyValueEntry(DummyStringKeys.DUMMY_3_NAME, true)
+                .addReport(createReport("sub report 1"))
+                .addReport(createReport("sub report 2"));
+
+        assertThat(reportBuilder).isInstanceOf(report.getReportBuilderClass());
+
+        Report builtReport = reportBuilder.build();
+
+        assertThatReport(builtReport)
+                .hasName("report")
+                .hasEntriesContaining(new StringEntry("entry"),
+                        new StringEntry("String Entry"),
+                        new KeyValueEntry("Key Entry", "Value Entry"),
+                        new FileEntry("file path"),
+                        new KeyValueEntry("key", "value"),
+                        new StringEntry("list_entries_1"),
+                        new StringEntry("list_entries_2"),
+                        new StringEntry("array_entries_1"),
+                        new StringEntry("array_entries_2"),
+                        new StringEntry("array_entries_1"),
+                        new StringEntry("array_entries_2"),
+                        new KeyValueEntry("Map_Key_1", "Map_Value_1"),
+                        new KeyValueEntry("Map_Key_2", "Map_Value_2"),
+                        new KeyValueEntry(DummyStringKeys.DUMMY_1_NAME, "value"),
+                        new KeyValueEntry(DummyStringKeys.DUMMY_2_NAME, "1"),
+                        new KeyValueEntry(DummyStringKeys.DUMMY_3_NAME, "true"))
+                .hasSubReportsContaining(Arrays.asList(new BasicReport("sub report 1"), new BasicReport("sub report 2")))
+                .hasNumberOfEntries(16)
+                .hasNumberOfSubReports(2)
+                .isInstanceOf(reportClass);
+    }
+
+    @Override
+    protected void addAdditionalExtensions(List<Class<?>> extensions) {
+
+    }
```

Comment:
inheritance - because we need to do some preparation/registration in before methods - the AbstractReporterTestBase extends AbstractTestTestBase from core and adds some additional functionality.
Or, I could change the method not to be abstract and leave it empty in the base class with an opportunity of overriding it.